index.html

```diff
@@ -23,13 +23,6 @@
     <meta name="theme-color" content="#000" />
     <title>em</title>
 
-    <!-- Set background color to black in dark mode to avoid white flash while CSS loads. -->
```

Comment:
This is confuses me, since I don't get a white flash locally, and the .dark shouldn't ever even be applied

Reply:
Actually I take that back, I do see this on first load.
I'd like to try adding this to <script>, but I'm not sure how to import `getTheme` and  `injectTheme`
```js
const pandaTheme = theme && (await getTheme(theme))
pandaTheme && injectTheme(document.documentElement, pandaTheme) // this returns the injected style element
```

Reply:
@raineorshine is there a way I can use `getTheme` and `injectTheme` inside index.html?

Reply:
I'm not sure. I'll let you and @trevinhofmann figure that out.

Reply:
Sounds good. @trevinhofmann I do have the styles in panda css `body: { overscrollBehaviorY: 'contain', color: 'fg', backgroundColor: 'bg' }`, I don't get any white flash from refreshing, only on first load, which does make some sense, considering it's the dev server. I tried the production build, and saw no white flashes, so I'm not sure where your white flash is coming from. Does it exist for you on the production build as well?

Reply:
one possible fix is using `document.body.style.backgroundColor = theme === 'dark' ? 'black' : 'white'`. I'm not sure what I feel about using the style prop though, since it can't be overriden by classes. Another is to just keep the background.dark, only for `index.html`

Reply:
It feels like a slight amount of tech debt to use `document.body.style.backgroundColor = theme === 'dark' ? 'black' : 'white'`, but I don't see any better/simpler alternatives for avoiding the flash. The theme is being assigned in `storageCache/theme`, and we already retrieve and use that in `index.html` here:

https://github.com/cybersemics/em/blob/6c9bef17209c38c94c447a4294c01c76d353830c/index.html#L49-L52

For that reason, I recommend we add it in that same place like this::

https://github.com/cybersemics/em/blob/7eb6b164daabb8b818c8d35442624fbfdafa62f1/index.html#L42-L43

I've added a commit to do that in this PR here: https://github.com/cybersemics/em/pull/2494/commits/7eb6b164daabb8b818c8d35442624fbfdafa62f1

Reply:
@raineorshine, after adding the above change, I noticed a very strange bug. Whenever I toggled the theme and refreshed, the initial value of `localStorage.getItem('storageCache/theme')` was incorrect and then corrected itself later after the page loaded, causing the body and app to have mismatched background colors (e.g. the page itself was dark, but the thoughtspace and other components were light).

I investigated further with some logging and noticed that the `storageCache/theme` was not being assigned at all during the first toggle in a single session, and then updated to the incorrect value with subsequent toggling. Despite that, the values in the theme "thought" were being correctly updated.

Continued digging led me to `storageCacheStoreEnhancer` as the likely source of the problem. Specifically, this part had the missing (and then incorrect) cache updates despite the correct state:

https://github.com/cybersemics/em/blob/6c9bef17209c38c94c447a4294c01c76d353830c/src/redux-enhancers/storageCache.ts#L86-L95

I believe this should be retrieving the state updates by looking at `newState` rather than the old value of `state`. I've added a commit to this PR here to apply that fix: https://github.com/cybersemics/em/pull/2494/commits/464666459d673f8985d5d9093053b997f95d1a42

Can you please check to make sure that this makes sense to you as well? I was also hesitant to add it to this PR since it's not exactly within scope, but it's quite a small fix and the issue in this thread depends on the fix.

Reply:
Good catch! It must have been the muscle memory of always typing `state` ☺️

The change on L87 to `const value = controller.select(stateNew)` makes sense to me and should fix the observed issue.

The change on L90 to `return valueParsed !== stateNew.storageCache?.[key as keyof State['storageCache']]` is trying to compare the new value with the old value, and it only works because `stateNew.storageCache` has not been updated yet. I think `state` probably makes more sense here, as it more explicitly describes that this is a comparison with the old state value. Please test though and correct me if I'm wrong!

Reply:
I [refactored](https://github.com/cybersemics/em/commit/c08526f889) to make it clearer and [fixed](https://github.com/cybersemics/em/commit/0d2d408466) the comparison. Let me know if that makes sense to you.

Reply:
LGTM, thanks!